src/ResourceManagement/Network/Microsoft.Azure.Management.Network.Fluent/ApplicationGatewaySslCertificateImpl.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+namespace Microsoft.Azure.Management.Network.Fluent
+{
+    using ApplicationGatewaySslCertificate.Definition;
+    using ApplicationGatewaySslCertificate.UpdateDefinition;
+    using Models;
+    using Resource.Fluent.Core;
+    using System.IO;
+    using Resource.Fluent.Core.ChildResourceActions;
+    using System;
+
+    /// <summary>
+    /// Implementation for ApplicationGatewaySslCertificate.
+    /// </summary>
+    ///GENTHASH:Y29tLm1pY3Jvc29mdC5henVyZS5tYW5hZ2VtZW50Lm5ldHdvcmsuaW1wbGVtZW50YXRpb24uQXBwbGljYXRpb25HYXRld2F5U3NsQ2VydGlmaWNhdGVJbXBs
+    internal partial class ApplicationGatewaySslCertificateImpl :
+        ChildResource<Models.ApplicationGatewaySslCertificateInner, ApplicationGatewayImpl, IApplicationGateway>,
+        IApplicationGatewaySslCertificate,
+        IDefinition<ApplicationGateway.Definition.IWithCreate>,
+        IUpdateDefinition<ApplicationGateway.Update.IUpdate>,
+        ApplicationGatewaySslCertificate.Update.IUpdate
+    {
+        ///GENMHASH:10A46E6A29805F84C3DCA9670AF5A0BC:C0847EA0CDA78F6D91EFD239C70F0FA7
+        internal ApplicationGatewaySslCertificateImpl(ApplicationGatewaySslCertificateInner inner, ApplicationGatewayImpl parent) : base(inner, parent)
+        {
+        }
+
+        #region Withers
+
+        ///GENMHASH:F2EE2F07B6724F1FF58043814706A8ED:DB3DCD6CAFDAF739E813082765CD3375
+        public ApplicationGatewaySslCertificateImpl WithPfxFromBytes(params byte[] pfxData)
+        {
+            string encoded = null;
+            if (pfxData != null)
+            {
+                encoded = Convert.ToBase64String(pfxData);
+            }
+            Inner.Data = encoded;
+            return this;
+        }
+
+        ///GENMHASH:079EAB40DB57656F562BDBA357A86C43:8892D33C89EB45AB71456048E4189668
+        public ApplicationGatewaySslCertificateImpl WithPfxPassword(string password)
+        {
+            Inner.Password = password;
+            return this;
+        }
+
+        ///GENMHASH:94F90A4A28B1E983315F51729B27DB77:71EBC245EFF014FF21D0CBEC5F53A54C
+        public ApplicationGatewaySslCertificateImpl WithPfxFromFile(FileInfo pfxFile)
+        {
+            if (pfxFile == null) {
+                return null;
+            }
+
+            try
+            {
+                byte[] content = File.ReadAllBytes(pfxFile.FullName);
+                return (content != null) ? WithPfxFromBytes(content) : null;
+            }
+            catch
+            {
+                return null;
```

Comment:
agreed on the specific case of letting the file-IO based exceptions go through.
(in general, as discussed, nulls vs exceptions on the withers are a bit more nuanced issue --- will address on a case by case basis)